evm/db/chain.py

```diff
@@ -312,11 +313,19 @@ def get_block_header_by_hash(self, block_hash: bytes) -> BlockHeader:
         """
         validate_word(block_hash, title="Block Hash")
         try:
-            block = self.db.get(block_hash)
+            header_rlp = self.db.get(block_hash)
         except KeyError:
             raise BlockNotFound("No block with hash {0} found".format(
                 encode_hex(block_hash)))
-        return rlp.decode(block, sedes=BlockHeader)
+        return self._decode_block_header(header_rlp)
+
+    # When performing a chain sync (either fast or regular modes), we'll very often need to look
+    # up recent block headers to validate the chain, and decoding their RLP representation is
+    # relatively expensive so we cache that here, but use a small cache because we *should* only
+    # be looking up recent blocks.
+    @functools.lru_cache(128)
+    def _decode_block_header(self, header_rlp: bytes) -> BlockHeader:
```

Comment:
Probably good to go ahead and wrap this in `staticmethod` *or* elevate it to a top level module function.

Reply:
Makes sense, went with a top-level function